helm/oauth2-proxy/templates/configmap.yaml

```diff
@@ -13,6 +13,13 @@ metadata:
   name: {{ template "oauth2-proxy.fullname" . }}
   namespace: {{ template "oauth2-proxy.namespace" $ }}
 data:
+  {{- if and (not .Values.alphaConfig.enabled) .Values.config.legacyDefaults }}
```

Comment:
Isn't legacyDefaults always defined through the default values in `values.yaml` and even if a user reconfigures it to be an empty object or empty string wouldn't the templating still work?

I'm just asking myself if we really need the if else condition here? Furthermore if required, I would rather prefer to have it surround the actual templating line for the legacyDefaults instead of the whole `oauth2_proxy.cfg` block

`{{ tpl .Values.config.legacyDefaults $ | indent 4 }}`

Reply:
That would probably work too, but I'm done w/ this one so someone else will have to pick it up.  If I was going to spend any more time on this I would just put the remaining option into alphaConfig so the whole legacyDefaults can go away. This isn't supposed to be a long lasting fix. Just a patch to get you to the next point.

Reply:
I appreciate the attempt to help, but after checking this change introduces a breaking change, and we can’t merge it as-is. I know it seems trivial but at the scale oauth2-proxy is being utilized (we are talk 30+ M downloads a month), even minor tweaks need thorough validation! Throwing in a patch and declaring you’re ‘done’ isn’t how stable software gets maintained.